airbyte-integrations/connectors/source-insightly/source_insightly/manifest.yaml

```diff
@@ -0,0 +1,390 @@
+version: "0.29.0"
+
+definitions:
+  selector:
+    type: RecordSelector
+    extractor:
+      type: DpathExtractor
+      field_path: []
+  requester:
+    type: HttpRequester
+    url_base: "https://api.na1.insightly.com/v3.1/"
+    http_method: "GET"
+    authenticator:
+      type: BasicHttpAuthenticator
+      username: "{{ config['token'] }}"
+    request_options_provider:
+      request_parameters:
+        count_total: True,
+  date_incremental_sync:
+    type: DatetimeBasedCursor
+    start_datetime:
+      datetime: "{{ config['start_date'] }}"
+      datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+    end_datetime:
+      datetime: "{{ now_utc() }}"
+      datetime_format: "%Y-%m-%d %H:%M:%S.%f+00:00"
+    datetime_format: "%Y-%m-%dT%H:%M:%S"
+    cursor_granularity: PT1S
+    step: P1D
+    cursor_field: "DATE_UPDATED_UTC"
+  retriever:
+    type: SimpleRetriever
+    record_selector:
+      $ref: "#/definitions/selector"
+    paginator:
+      type: DefaultPaginator
+      pagination_strategy:
+        type: "OffsetIncrement"
+        page_size: 500
+      page_token_option:
+        type: RequestOption
+        inject_into: "request_parameter"
+        field_name: "skip"
```

Comment:
I also want to highlight this part regarding the pagination. Insightly has the skip parameter which skips a certain amount of records. For example if the first request gives us 500 records then for the next request we will add skip=500 to the parameter, however for the next request after it we will have to add skip=100, i.e the skip gets incremented so that all records are parsed. How do we apply it here?

Reply:
Did you try to use the Offset paginator?